pkg/util/provider/machinecontroller/machine.go

```diff
@@ -359,14 +354,39 @@ func (c *controller) triggerCreationFlow(ctx context.Context, createMachineReque
 
 				nodeName = createMachineResponse.NodeName
 				providerID = createMachineResponse.ProviderID
+
+				// Creation was successful
+				klog.V(2).Infof("Created new VM for machine: %q with ProviderID: %q and backing node: %q", machine.Name, providerID, getNodeName(machine))
+
+				//if a stale node obj exists by the same nodeName
+				if _, err := c.nodeLister.Get(nodeName); err == nil {
+					//mark the machine obj as `Failed`
+					klog.Errorf("Stale node obj with name %q has been found", nodeName)
+
+					c.machineStatusUpdate(
```

Comment:
We are updating the status of the machine without really updating the `providerID`. So the VM deletion will depend on the fact that we have tagged the VMs properly. 

The better way would be to delete the created VM at this step here, before updating the status. Or have the providerID updated so that deletion has the providerID? WDYT?